### PR TITLE
[Websearch Action] Handle serpApi errors & no results

### PR DESCRIPTION
**Pull request by philipperolet**
Description
---
When serpApi finds no results, it returns an error. This lead to a cryptic failure such as [here](https://dust.tt/w/0ec9852c2f/assistant/b43235c58c).

This PR handles the no-results case (which should not err), and any other error case (which should err)

Also, using the opportunity to io-ts WebsearchActionOutputType so we get proper errors in weird cases (vs the previous `as`)

Risk
---
N/A (gated)

Deploy
---
Front

### CODE REVIEW

File: types/src/front/assistant/actions/websearch.ts [hunk 1]
```diff
@@ -10,15 +12,27 @@ export type WebsearchConfigurationType = {
...
+  snippet: t.string,
+  link: t.string,
+});
 
-export type WebsearchResultType = {
-  title: string;
-  snippet: string;
-  link: string;
-};
+export const WebsearchActionOutputSchema = t.union([
```
Comment (PopDaph) [inline types/src/front/assistant/actions/websearch.ts:21]: NIT: It does not work to declare error as string or undefined? or partial?  (so we don't duplicate `results`)

File: front/components/assistant/conversation/WebsearchAction.tsx [hunk 2]
```diff
@@ -25,6 +26,19 @@ export default function WebsearchAction({
 
   const { results } = websearchAction.output ?? { results: [] };
 
+  const rawError =
+    websearchAction.output && "error" in websearchAction.output
```
Comment (PopDaph) [inline front/components/assistant/conversation/WebsearchAction.tsx:30]: NIT: shouldn't you update `WebsearchActionOutputType` to set that there's an optional error string so this code could be simplified?

**Review by PopDaph — APPROVED** [review]
Approving already, my comments are non blocking 🙏🏻

File: types/src/front/assistant/actions/websearch.ts (hunk 1 above)
Comment (philipperolet) [reply types/src/front/assistant/actions/websearch.ts:21]: unfortunately no 😔 (I couldn't make it work nicely, it refused calls with results but no error)

File: front/components/assistant/conversation/WebsearchAction.tsx (hunk 2 above)
Comment (philipperolet) [reply front/components/assistant/conversation/WebsearchAction.tsx:30]: can't make an optional field with io-ts unfortunately AFAIK (after having tried)

**Pull request merged**